### PR TITLE
refactor: Block .head consistently in unordered mode

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/core/groupby/__init__.py [hunk 1]
```diff
@@ -109,6 +109,7 @@ def __getitem__(
                 dropna=self._dropna,
             )
 
+    @validations.requires_strict_ordering()
```
Comment (shobsi) [inline bigframes/core/groupby/__init__.py:112]: do we plan to implement peek() in groupby classes later and then suggest that here when strict ordering is disabled?

File: bigframes/series.py [hunk 2]
```diff
@@ -633,7 +633,7 @@ def dropna(
             result = result.reset_index()
         return Series(result)
 
-    @validations.requires_strict_ordering()
+    @validations.requires_strict_ordering("Use .peek(n) to preview n arbitrary rows.")
```
[the remark this answers is not in the document]
Comment (TrevorBergeron) [reply bigframes/series.py]: Extracted to constant

File: bigframes/core/groupby/__init__.py (hunk 1 above)
Comment (TrevorBergeron) [reply bigframes/core/groupby/__init__.py:112]: No, unlikely we will add peek() to groupby classes, as you can't really do an efficient limit clause per-group in bigquery.

File: bigframes/dataframe.py [hunk 3]
```diff
@@ -1293,6 +1293,7 @@ def _compute_dry_run(self) -> bigquery.QueryJob:
     def copy(self) -> DataFrame:
         return DataFrame(self._block)
 
+    @validations.requires_strict_ordering("Use .peek(n) to preview n arbitrary rows.")
```
Comment (shobsi) [inline bigframes/dataframe.py:1296]: missed reusing the const here

File: bigframes/series.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (shobsi) [reply bigframes/series.py]: Thanks!

File: bigframes/dataframe.py (hunk 3 above)
Comment (TrevorBergeron) [reply bigframes/dataframe.py]: oops, fixed in new revision

**Review by shobsi — APPROVED** [review]

**Pull request merged**